core/src/main/scala/kafka/log/streamaspect/ElasticLogSegmentManager.java

```diff
@@ -164,6 +170,44 @@ public void onEvent(long segmentBaseOffset, ElasticLogSegmentEvent event) {
                 }
             }
         }
+
+        private void submitOrDrainPendingPersistentMetaQueue(long segmentBaseOffset) {
+            if (segmentBaseOffset != NO_OP_OFFSET) {
+                pendingDeleteSegmentBaseOffset.add(segmentBaseOffset);
+            }
+
+            if (pendingPersistentMetaCf != null && !pendingPersistentMetaCf.isDone()) {
```

Comment:
pendingPersistentMetaCf should also guard by synchronized

Reply:
code changed ~